In [ ]:
import math

12. How many bit strings of length 12 contain :

    a) exactly three 1s?

    b) at most three 1s?

    c) at least three 1s?

    d) an equal number of 0s and 1s?
    
    Jawab :

**a** :

Dik : n = 12 dan r = 3

In [ ]:
def kombinasi(n,r):
  return math.factorial(n)//(math.factorial(r)*math.factorial(n-r))

In [ ]:
hasil_a = kombinasi(12,3)
print(hasil_a)

220


Jadi, ada 220 string biner dengan panjang 12 yang angka 1 nya tepat ada 3

**b** :

Dik : n = 12 dan r = 0 sampai 3

In [ ]:
def kombinasi(n,r):
  return math.factorial(n)//(math.factorial(r)*math.factorial(n-r))

In [ ]:
# Ini untuk r = 0
hasil_b = kombinasi(12,0)
print(hasil_b)

1


In [ ]:
# Ini untuk r = 1
hasil_b = kombinasi(12,1)
print(hasil_b)

12


In [ ]:
# Ini untuk r = 2
hasil_b = kombinasi(12,2)
print(hasil_b)

66


In [ ]:
# Ini untuk r = 3
hasil_b = kombinasi(12,3)
print(hasil_b)

220


In [ ]:
# Menjumlahkan semua hasilnya
hasil_akhir_b = 1 + 12 + 66 + 220
print("Hasil akhir = ", hasil_akhir_b)

Hasil akhir =  299


Jadi, ada 299 string biner dengan panjang 12 yang angka 1 nya maksimal ada 3

**c** :

Dik : n = 12 dan r = 3 sampai 12

In [ ]:
def kombinasi(n,r):
  return math.factorial(n)//(math.factorial(r)*math.factorial(n-r))

In [ ]:
# Ini untuk r = 3
hasil_c = kombinasi(12,3)
print(hasil_c)

220


In [ ]:
# Ini untuk r = 4
hasil_c = kombinasi(12,4)
print(hasil_c)

495


In [ ]:
# Ini untuk r = 5
hasil_c = kombinasi(12,5)
print(hasil_c)

792


In [ ]:
# Ini untuk r = 6
hasil_c = kombinasi(12,6)
print(hasil_c)

924


In [ ]:
# Ini untuk r = 7
hasil_c = kombinasi(12,7)
print(hasil_c)

792


In [ ]:
# Ini untuk r = 8
hasil_c = kombinasi(12,8)
print(hasil_c)

495


In [ ]:
# Ini untuk r = 9
hasil_c = kombinasi(12,9)
print(hasil_c)

220


In [ ]:
# Ini untuk r = 10
hasil_c = kombinasi(12,10)
print(hasil_c)

66


In [ ]:
# Ini untuk r = 11
hasil_c = kombinasi(12,11)
print(hasil_c)

12


In [ ]:
# Ini untuk r = 12
hasil_c = kombinasi(12,12)
print(hasil_c)

1


In [ ]:
# Menjumlahkan semua hasilnya
hasil_akhir_c = 220 + 495 + 792 + 924 + 792 + 495 + 220 + 66 + 12 + 1
print("Hasil akhir = ", hasil_akhir_c)

Hasil akhir =  4017


Jadi,  ada 4.017 string biner dengan panjang 12 yang angka 1 minimal ada 3

**d** :

Dik : n = 12 dan r = 6

In [ ]:
def kombinasi(n,r):
  return math.factorial(n)//(math.factorial(r)*math.factorial(n-r))

In [ ]:
hasil_d = kombinasi(12,6)
print(hasil_d)

924


Jadi, ada 924 string biner dengan panjang 12 yang angka 0 dan 1 nya jumlahnya sama

Ringkasan jawaban nomor 12 :

a). 220

b). 299

c). 4.017

d). 924

 -------------------------------------------------------------------------------

13. A group contains n men and n women. How many ways
are there to arrange these people in a row if the men and
women alternate?

    Jawab :

    Diketahui ada n! cara untuk menyusun n pria dan n! cara untuk n menyusun wanita, maka setiap pola jumlah penyusunannya adalah n! * n! atau n!^2.

    Karena r nya atau pola nya ada 2 (pola pertama pria yang duluan dan pola kedua wanita yang duluan), maka total cara penyusunannya adalah :

    r * (n!)^2

    = 2 * (n!)^2

In [ ]:
# Kita anggap n nya adalah 5
n = 5
r = 2

cara_penyusunan = r*(math.factorial(n)**2)
print(cara_penyusunan)

28800


Jadi, banyaknya cara untuk menyusun orang-orang itu jika pria dan wanita harus berselang-seling adalah sebanyak 28.800 cara

--------------------------------------------------------------------------------

14. In how many ways can a set of two positive integers less
than 100 be chosen?

    Jawab :

    Dik : n = 99 (karena kurang dari 100) dan r = 2

In [ ]:
def kombinasi(n,r):
  return math.factorial(n)//(math.factorial(r)*math.factorial(n-r))

In [ ]:
hasil = kombinasi(99,2)
print(hasil)

4851


Jadi, ada 4.851 cara untuk memilih 2 bilangan bulat positif yang kurang dari 100

--------------------------------------------------------------------------------

15. In how many ways can a set of five letters be selected
from the English alphabet?

    Jawab :

    Dik : n = 26 (karena jumlah alfabet ada 26) dan r = 5

In [ ]:
def kombinasi(n,r):
  return math.factorial(n)//(math.factorial(r)*math.factorial(n-r))

In [ ]:
hasil = kombinasi(26,5)
print(hasil)

65780


Jadi, ada 65.780 cara untuk memilih 5 huruf dari alfabet

--------------------------------------------------------------------------------

16. How many subsets with an odd number of elements does
a set with 10 elements have?

    Jawab :
    
    Dik : n = 10

    Jumlah total subset dari himpunan dengan n elemen adalah :

    2^n

In [ ]:
n = 10

total_jumlah = 2**n
print(total_jumlah)

1024


Dari 1.024 subset ini, setengahnya akan memiliki jumlah elemen ganjil dan setengahnya lagi genap, maka harus dibagi 2 dulu

In [ ]:
subset_dengan_jumlahElementnyaGanjil = total_jumlah//2
print(subset_dengan_jumlahElementnyaGanjil)

512


Jadi, ada 512 subset dengan jumlah elemennya ganjil.

--------------------------------------------------------------------------------

17. How many subsets with more than two elements does a
set with 100 elements have?

    Jawab :

    Dik : n = 100

    Jumlah total subset dari himpunan dengan n elemen adalah :

    2^n

In [70]:
n = 100

total_subset = 2**n
print(f"{total_subset:,}")

1,267,650,600,228,229,401,496,703,205,376


Kemudian kita mengurangi total subset itu dengan subset yang memiliki 2 elemen atau kurang (berarti r = 0 sampai 2)

In [71]:
def kombinasi(n,r):
  return math.factorial(n)//(math.factorial(r)*math.factorial(n-r))

In [72]:
# Ini untuk r = 0
hasil = kombinasi(100,0)
print(hasil)

1


In [73]:
# Ini untuk r = 1
hasil = kombinasi(100,1)
print(hasil)

100


In [74]:
# Ini untuk r = 2
hasil = kombinasi(100,2)
print(hasil)

4950


In [75]:
# Menjumlahkan hasilnya
jumlah_hasil = 1 + 100 + 4950
print(jumlah_hasil)

5051


Sekarang kita mengurangi total subset dengan 5.051

In [76]:
subset_YangMemiliki_LebihDari_2Elemen = total_subset - jumlah_hasil
print(f"{subset_YangMemiliki_LebihDari_2Elemen:,}")

1,267,650,600,228,229,401,496,703,200,325


Jadi, jumlah subset yang memiliki elemen lebih dari 2 dengan 100 elemen adalah : 1.267.650.600.228.229.401.496.703.200.325 (banyak yah)

--------------------------------------------------------------------------------

18. A coin is flipped eight times where each flip comes up
either heads or tails. How many possible outcomes

    a) are there in total?

    b) contain exactly three heads?

    c) contain at least three heads?

    d) contain the same number of heads and tails?

    Jawab :

**a** :

Dik : n = 8

Total kemungkinan hasil adalah :

2^n

In [ ]:
n = 8

total_kemungkinan_hasil = 2**n
print(total_kemungkinan_hasil)

256


Jadi, terdapat 256 kemungkinan hasilnya secara keseluruhan

**b** :

Dik : n = 8 dan r = 3

In [ ]:
def kombinasi(n,r):
  return math.factorial(n)//(math.factorial(r)*math.factorial(n-r))

In [ ]:
hasil = kombinasi(8,3)
print(hasil)

56


Jadi, terdapat 56 hasil yang mengandung tepat tiga heads

**c** :

Dik : n = 8 dan r = 3 sampai 8

In [ ]:
def kombinasi(n,r):
  return math.factorial(n)//(math.factorial(r)*math.factorial(n-r))

In [ ]:
# Ini untuk r = 3
hasil = kombinasi(8,3)
print(hasil)

56


In [ ]:
# Ini untuk r = 4
hasil = kombinasi(8,4)
print(hasil)

70


In [ ]:
# Ini untuk r = 5
hasil = kombinasi(8,5)
print(hasil)

56


In [ ]:
# Ini untuk r = 6
hasil = kombinasi(8,6)
print(hasil)

28


In [ ]:
# Ini untuk r = 7
hasil = kombinasi(8,7)
print(hasil)

8


In [ ]:
# Ini untuk r = 8
hasil = kombinasi(8,8)
print(hasil)

1


In [ ]:
# Menjumlahkan hasilnya
jumlah_hasil = 56 + 70 + 56 + 28 + 8 + 1
print(jumlah_hasil)

219


Jadi, terdapat 219 hasil dengan setidaknya tiga heads

**d** :

Dik : n = 8 dan r = 4

In [ ]:
def kombinasi(n,r):
  return math.factorial(n)//(math.factorial(r)*math.factorial(n-r))

In [ ]:
# Ini untuk r = 4
hasil = kombinasi(8,4)
print(hasil)

70


Jadi, terdapat 70 hasil dengan jumlah heads dan tails yang sama

Ringkasan jawaban nomor 18 :

a). 256

b). 56

c). 219

d). 70